pkg/compose/layers.go

```diff
@@ -237,12 +237,20 @@ func ComposeAppLayersManifest(arch string, layers []distribution.Descriptor) (di
 	if err != nil {
 		return nil, nil, err
 	}
-	man, desc, err := distribution.UnmarshalManifest(v1.MediaTypeImageIndex, manifestJson)
+
+	// Convert the custom manifest that breaks the OCI image specification into the type that the distribution
+	// functionality understands. Use the json unmarshalling functionality for it.
+	dm := manifestlist.DeserializedManifestList{}
+	err = dm.UnmarshalJSON(manifestJson)
 	if err != nil {
 		return nil, nil, err
 	}
-	desc.Platform = &platform
-	return man, &desc, nil
+	return dm, &distribution.Descriptor{
+		MediaType: dm.MediaType,
+		Size:      int64(len(manifestJson)),
+		Digest:    digest.FromBytes(manifestJson),
+		Platform:  manifestDef.Platform,
+	}, nil
```

Comment:
to make sure I'm reading this correctly - the manifest is the same, you are just generating it in a different way?

Reply:
Yes. This is another way to generate the same manifest; this way makes it less dependent on weirdness of the distribution/registry code.